### PR TITLE
create a task to load the other devices one by one

### PR DESCRIPTION
**Pull request by wgzintel**
### Details:
 - *create a task to load the other devices one by one*

### Tickets:
 - *ticket-75212*

### CODE REVIEW

File: src/plugins/auto/executable_network.cpp [hunk 1]
```diff
@@ -504,6 +603,19 @@ bool MultiDeviceExecutableNetwork::ScheduleToWorkerInferRequest(Task inferPipeli
                 deviceInfo.deviceName = _loadContext[CPU].workName;
                 devices.push_back(std::move(deviceInfo));
             }
+
+            if (_context.performanceHint == PluginConfigParams::CUMULATIVE_THROUGHPUT) {
+                for (auto deviceInfo : _loadContext[ACTUALDEVICE].cumulativeDevices) {
```
Comment (tiger100256-hu) [inline src/plugins/auto/executable_network.cpp:608]: cumulativeDevices is not locked

Comment (songbell) [reply src/plugins/auto/executable_network.cpp:608]: also please make sure the device order is kept as it was

File: samples/cpp/benchmark_app/main.cpp [hunk 2]
```diff
@@ -56,9 +56,9 @@ bool parse_and_check_command_line(int argc, char* argv[]) {
         throw std::logic_error("Incorrect API. Please set -api option to `sync` or `async` value.");
     }
     if (!FLAGS_hint.empty() && FLAGS_hint != "throughput" && FLAGS_hint != "tput" && FLAGS_hint != "latency" &&
-        FLAGS_hint != "none") {
+        FLAGS_hint != "cumulative_throughput" && FLAGS_hint != "none") {
```
Comment (songbell) [inline samples/cpp/benchmark_app/main.cpp:59]: maybe we should have a short cut for this string also

Comment (peterchen-intel) [reply samples/cpp/benchmark_app/main.cpp:59]: ctput

File: samples/cpp/benchmark_app/main.cpp [hunk 3]
```diff
@@ -119,6 +119,9 @@ ov::hint::PerformanceMode get_performance_hint(const std::string& device, const
             } else if (FLAGS_hint == "latency") {
                 slog::warn << "Device(" << device << ") performance hint is set to LATENCY" << slog::endl;
                 ov_perf_hint = ov::hint::PerformanceMode::LATENCY;
+            } else if (FLAGS_hint == "cumulative_throughput") {
```
Comment (tiger100256-hu) [inline samples/cpp/benchmark_app/main.cpp:122]: FLAGS_hint == "cumulative_throughput" || FLAGS_hint == "ctput"

File: src/plugins/auto/executable_network.cpp [hunk 4]
```diff
@@ -407,33 +525,36 @@ void MultiDeviceExecutableNetwork::TryToLoadNetWork(AutoLoadContext& context,
...
+    if (IsCpuLoaded(context.deviceInfo, _loadContext[CPU].deviceInfo.config)) {
+        return;
+    }
+
+    LOG_DEBUG("[AUTOPLUGIN] try to load %s", context.deviceInfo.deviceName.c_str());
+    // try to load this candidate device
+    TryToLoadNetWork(context, modelPath, network);
+}
+
+bool MultiDeviceExecutableNetwork::IsCpuLoaded(DeviceInformation devInfo, std::map<std::string, std::string>& b) {
```
Comment (tiger100256-hu) [inline src/plugins/auto/executable_network.cpp:537]: const DeviceInformation &

File: src/plugins/auto/executable_network.cpp (hunk 1 above)
Comment (tiger100256-hu) [reply src/plugins/auto/executable_network.cpp:634]: push ACTUALDEVICE; then push cumulativeDevices; at last if not exist CPU_HELP in devices  try to push CPU_HELP

File: src/plugins/auto/executable_network.cpp [hunk 5]
```diff
@@ -809,77 +954,42 @@ InferenceEngine::Parameter MultiDeviceExecutableNetwork::GetMetric(const std::st
...
-                bool bThroughputEnabledInPlugin = false;
-                try {
-                    // for benchmark through AUTO:CPU,GPU
-                    // SetConfig directly set to CPU/GPU in this case
-                    bThroughputEnabledInPlugin =
-                        _core->GetConfig(deviceInfo.deviceName, CONFIG_KEY(PERFORMANCE_HINT)).as<std::string>() == CONFIG_VALUE(THROUGHPUT);
-                } catch (...) {
-                    LOG_DEBUG("[AUTOPLUGIN]GetMetric:%s for %s", "PERF_HINT config not supported", deviceInfo.deviceName.c_str());
+            if (_context.performanceHint == PluginConfigParams::CUMULATIVE_THROUGHPUT) {
+                for (auto& device : _loadContext[ACTUALDEVICE].metaDevices) {
```
Comment (tiger100256-hu) [inline src/plugins/auto/executable_network.cpp:958]: metaDevices is not safe here

File: src/plugins/auto/executable_network.cpp [hunk 6]
```diff
@@ -251,23 +261,27 @@ MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const std::string&
          }
     }
 
-    OV_ITT_SCOPED_TASK(itt::domains::MULTIPlugin, openvino::itt::handle(profilingTask));
-    if (_loadContext[CPU].isEnabled) {
-        _firstLoadFuture = _firstLoadPromise.get_future();
+    //get Idle CPU Streams Executor
+    if (_loadContext[CPU].isEnabled || isCumulative) {
```
Comment (songbell) [inline src/plugins/auto/executable_network.cpp:265]: double check if -d AUTO:GPU.0,GPU.1 cumulative works as expected?

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**